frame/executive/src/lib.rs

```diff
@@ -360,12 +362,32 @@ where
 		sp_tracing::enter_span!( sp_tracing::Level::TRACE, "finalize_block" );
 		<frame_system::Module<System>>::note_finished_extrinsics();
 		let block_number = <frame_system::Module<System>>::block_number();
-		<frame_system::Module<System> as OnFinalize<System::BlockNumber>>::on_finalize(block_number);
-		<AllModules as OnFinalize<System::BlockNumber>>::on_finalize(block_number);
+
+		Self::idle_and_finalize_hook(block_number);
```

Comment:
it is a little strange to me you have coupled these two together

Reply:
ya, at first there were just two lines of duplicate code, then after adding the hook, there was 8, so I thought it was prudent to handle all of them. I was thinking ok both hooks, they can go in the same function block. If you feel strongly that I should decouple them, I can make an on idle function and on finalize one.